### PR TITLE
MRG: Remove misleading info["filename"]

### LINKED ISSUE #3855: ica.info["filename"] ?
If I understand it correctly, `ica.info["filename"]` can be
- the name of the ica-fif - file on disk, if the ICA object was read from disk
- the name of the data file used for fitting

This is 1. a bit confusing, 2. potentially dangerous if you use something like

`ica.save(ica.info["filename"])`

See what I mean?

I think it should either 1. always default to the file used for fitting, or 2. to None if not read from disk and to the file name if read from disk.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #3855.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
CIs are happy

File: doc/cited.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+:orphan:
```
Comment (Eric89GXL) [inline doc/cited.rst]: These make it so that sphinx does not complain about files not existing in toctrees. We link to them manually outside of toctrees.

**Comment by agramfort** [comment]
LGTM

another pair of eyes?

@jona-sassenhagen are you ok with this fix?

**Comment by agramfort** [comment]
@Eric89GXL you need to rebase.

@jona-sassenhagen please confirm that it's ok for you.

**Comment by Eric89GXL** [comment]
Rebased

**Comment by jona-sassenhagen** [comment]
I can't evaluate the full PR because it touches on code I don't know, but what I get looks good.

**Comment by agramfort** [comment]
all good !

**Pull request merged**